presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryProvider.java

```diff
@@ -75,7 +92,23 @@ public DictionaryResult getDictionary(StreamDescriptor streamDescriptor, long[]
             throws IOException
     {
         InputStreamSource<LongInputStream> dictionaryDataStream = dictionaryStreamSources.getInputStreamSource(streamDescriptor, DICTIONARY_DATA, LongInputStream.class);
-        return loadDictionary(streamDescriptor, requireNonNull(dictionaryDataStream), dictionary, items);
+
+        if (streamDescriptor.getSequence() == 0 || requireNonNull(dictionaryDataStream).openStream() != null) {
```

Comment:
Please don't inline requireNonNull into the if-statement, it significantly reduces readability.

Do we care about the returned dictionaryDataStream.openStream?

Reply:
We do. In fact, that's the gist of the fallback mechanism. Curious actually, if we don't need to inline, why do we have the signature for requireNonNull? Why not just have a dedicated check not null statement?

Reply:
> why do we have the signature for requireNonNull?
Different teams have different requirements, for example in this case I try to avoid two levels of nesting. 
Having requireNonNull return the input argument is very when assigning fields in constructors.